app/src/main/java/com/besome/sketch/editor/view/ViewPane.java

```diff
@@ -419,122 +435,150 @@ private void updateItemView(View view, ViewBean viewBean) {
-                    }else {
-                        Bitmap decodeFile3 = BitmapFactory.decodeFile(imagelocation);
-                        ((ImageView) view).setImageBitmap(Bitmap.createScaledBitmap(decodeFile3, decodeFile3.getWidth() * round3, decodeFile3.getHeight() * round3, true));
+            if(viewBean.image.resName != null) {
+                if (view instanceof ImageView) {
+                    if (resourcesManager.h(viewBean.image.resName) == ProjectResourceBean.PROJECT_RES_TYPE_RESOURCE) {
+                        ((ImageView) view).setImageResource(getContext().getResources().getIdentifier(viewBean.image.resName, "drawable", getContext().getPackageName()));
+                    } else if (viewBean.image.resName.equals("default_image")) {
+                        ((ImageView) view).setImageResource(R.drawable.default_image);
+                    } else {
+                        try {
+                            String imagelocation = resourcesManager.f(viewBean.image.resName);
+
+                            int round3 = Math.round(getResources().getDisplayMetrics().density / 2.0f);
+                            if (imagelocation.endsWith(".xml")) {
+                                FilePathUtil fpu = new FilePathUtil();
+                                svgUtils.loadScaledSvgIntoImageView( (ImageView) view,fpu.getSvgFullPath(sc_id,viewBean.image.resName),round3);
+                            } else {
+                                Bitmap decodeFile3 = BitmapFactory.decodeFile(imagelocation);
+                                ((ImageView) view).setImageBitmap(Bitmap.createScaledBitmap(decodeFile3, decodeFile3.getWidth() * round3, decodeFile3.getHeight() * round3, true));
+                            }
+                        } catch (Exception unused2) {
+                            ((ImageView) view).setImageResource(R.drawable.default_image);
+                        }
+                    }
+                }
+                if (view instanceof ItemCircleImageView) {
+                    if (classInfo.b("CircleImageView")) {
+                        updateCircleImageView((ItemCircleImageView) view, injectHandler);
+                    } else {
+                        ((ImageView) view).setScaleType(ImageView.ScaleType.valueOf(viewBean.image.scaleType));
                     }
-                } catch (Exception unused2) {
-                    ((ImageView) view).setImageResource(R.drawable.default_image);
                 }
-            }
-            if (classInfo.b("CircleImageView")) {
-                updateCircleImageView((ItemCircleImageView) view, injectHandler);
-            } else {
-                ((ImageView) view).setScaleType(ImageView.ScaleType.valueOf(viewBean.image.scaleType));
             }
         }
         if (classInfo.a("CompoundButton")) {
-            ((CompoundButton) view).setChecked(viewBean.checked != 0);
+            if (view instanceof CompoundButton) {
```

Comment:
are these extra checks necessary? I think the views would already be of proper types

Reply:
well that fixed some  crashs related with cannot be cast

Reply:
Make it clear, crash that caused by changing the type of viewbean